### ISSUE TITLE
[HOLD for payment 2023-09-29] [$500] There is a larger gap between the Apple icon and the Google icon on Android

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#26566]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to login page
## Expected Result: 
The gap should be consistent across all platforms
## Actual Result: 
The gap is larger on Android

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.62-1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

![ef2d595a-cee9-47fa-bcdb-0b188f61bc9a](https://github.com/Expensify/App/assets/43996225/ce4a89b3-9bd3-4013-857b-e84c4b27d806)

![ca550428-8828-44e8-8291-37a1416f9f0c](https://github.com/Expensify/App/assets/43996225/5a70441f-6149-4b35-a492-52082e89c0ee)

**Expensify/Expensify Issue URL:**
**Issue reported by:** @aman-atg
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1693020637974179

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0158363a81364181da</li>
        <li>Upwork Job ID: 1698574418489659392</li>
        <li>Last Price Increase: 2023-09-04</li>
<li>Automatic offers: </li>
<ul>
<li>aman-atg | Contributor | 26711713</li>
<li>aman-atg | Reporter | 26711715</li>
</ul></li>
    </ul>
</details>

**Comment by kavimuru** [comment]
## Proposal from @aman-atg 
### Please re-state the problem that we are trying to solve in this issue.
There is a larger gap between the Apple icon and the Google icon on Android

### What is the root cause of that problem?
The root cause of the problem is the `styles` provided to `IconButton` [here](https://github.com/Expensify/App/blob/a6a05722631a016ec502611f663b8d1d7ae38d3e/src/components/SignInButtons/IconButton.js#L39) as it provides `margin:10px` for both buttons resulting in a larger gap in native devices since IconButton is being used there.


### What changes do you think we should make in order to solve the problem?

We should make changes [here](https://github.com/Expensify/App/blob/a6a05722631a016ec502611f663b8d1d7ae38d3e/src/styles/styles.js#L3655-L3659) to avoid horizontal margin for both buttons. Then, we can use `gap: 12` for button containers or `marginLeft: 12` for Google Button on all platforms to have consistent spacing.

### What alternative solutions did you explore? (Optional)

**Comment by sophiepintoraetz** [comment]
@kavimuru - please make sure you're checking all the affected platforms.
<img width="1937" alt="image" src="https://github.com/Expensify/App/assets/42940078/62d969c3-4436-4592-8e5f-705d4ee40825">

**Comment by payal-lathidadiya** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
There is a smaller gap between the Apple icon and the Google icon on Web/mWeb

### What is the root cause of that problem?
The cause of this inconsistent UI layout is due to different styles applied for component/elements in web and mWeb.

- On mWeb and native app, it renders as IconButton [here](https://github.com/Expensify/App/blob/main/src/components/SignInButtons/AppleSignIn/index.ios.js#L45). And it applies [these](https://github.com/Expensify/App/blob/main/src/styles/styles.js#L3655) CSS style
- On web, the same style is not applied for web component [here](https://github.com/Expensify/App/blob/main/src/components/SignInButtons/AppleSignIn/index.website.js#L93)

This resulted in inconsistency of margin/padding across platforms.

### What changes do you think we should make in order to solve the problem?

To fix this inconsistency, we can replace below style of web component to have same as sign in IconButton of mWeb/native [here](https://github.com/Expensify/App/blob/main/src/components/SignInButtons/AppleSignIn/index.website.js#L93) and [here](https://github.com/Expensify/App/blob/main/src/components/SignInButtons/AppleSignIn/index.website.js#L104)

```
style={styles.signInIconButton} 
```

This changes should be applied to both Google and Apple sign in components

### What alternative solutions did you explore? (Optional)
None

**Comment by sophiepintoraetz** [comment]
Waiting on @robertKozik to review the proposals!

**Comment by robertKozik** [comment]
Hi all! Thank you for your proposals 🙇🏼

It appears that all of you have accurately identified the root cause of the issue. Furthermore, all the proposed solutions were quite similar. However, I would prefer the solution from @aman-atg, as it moves the setting of `gap` to a file that is not platform-dependent. This way, we can ensure consistency without style repetition.

Selected Proposal: [proposal](https://github.com/Expensify/App/issues/26566#issuecomment-1703899140)
contributor: @aman-atg 

🎀 👀 🎀 C+ reviewed

**Comment by sophiepintoraetz** [comment]
@marcochavezf - please can you review the proposal on this issue?

**Comment by marcochavezf** [comment]
Thanks @robertKozik for reviewing the proposals. I agree with what you commented, assigning here @aman-atg 🚀

**Comment by marcochavezf** [comment]
Hmm it seems I cannot assign @aman-atg, is this the first GH issue that you will be assigned as contributor?

**Comment by sophiepintoraetz** [comment]
Am able to do so!

**Comment by thienlnam** [comment]
QA found that this was not fixed in Android https://github.com/Expensify/App/issues/27837#issuecomment-1728503025

**Comment by robertKozik** [comment]
@aman-atg Yeah, I think you're right. We overlooked this one during the initial PR. But maybe let's stick to changing the padding to be only vertical. This way, we won't introduce any change on the vertical axis.

**Comment by robertKozik** [comment]
Not awaiting payment yet. Followup PR is under review

**Comment by sophiepintoraetz** [comment]
Payment date is elapsed - paid out!
